flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamConfig.java

```diff
@@ -795,6 +795,20 @@ public boolean isGraphContainingLoops() {
         return config.getBoolean(GRAPH_CONTAINING_LOOPS, false);
     }
 
+    /**
+     * In general, we don't clear any configuration. However, the {@link #SERIALIZED_UDF} may be
+     * very large when operator includes some large objects, the SERIALIZED_UDF is used to create a
+     * StreamOperator and usually only needs to be called once. {@link #CHAINED_TASK_CONFIG} may be
+     * large as well due to the StreamConfig of all non-head operators in OperatorChain will be
+     * serialized and stored in CHAINED_TASK_CONFIG. They can be cleared to reduce the memory after
+     * StreamTask is initialized. If so, TM will have more memory during running. See FLINK-33315
+     * and FLINK-33317 for more information.
+     */
+    public void clearInitialConfigs() {
+        config.removeKey(SERIALIZED_UDF);
+        config.removeKey(CHAINED_TASK_CONFIG);
```

Comment:
nit: can we mark removal of those keys with some special value instead, so that we could throw a meaningful error in case of someone trying to read an already cleared value?

Reply:
Thanks @pnowojski for the quick response!

It's definitely a good suggestion, throwing a error is easy to let developer find this improvement if any developer get them in the future.

How about adding a `private final transient Set<String> removedKeys` to save them? If so, we don't need to get bytes from config and check whether it's removedValue in advance.

Of course, these 2 solution are similar, if you prefer store a special value, I can accept it.

Reply:
Both are fine by me :)

Reply:
Thanks @pnowojski for the feedback, updated.